File: 819. Most Common Word.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        String[] p = paragraph.toLowerCase().split("[ !?',;.]+");
        Map<String, Integer> map = new HashMap<>();
        for (String s : p) {
            map.put(s, map.getOrDefault(s, 0)+1);
        }
        for (String b : banned) {
            map.remove(b);
        }
        int count = 0;
        String word = "";
        for (String w : map.keySet()) {
            int temp = map.get(w);
            if (temp > count) {
                count = temp;
                word = w;
            }
        }
        return word;
    }
}

File: 819. Most Common Word.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        words = re.split("[ !?',;.]+", paragraph)
        dic = collections.defaultdict(int)
        ban = set(banned)
        for word in words:
            word = word.lower()
            if word and word not in ban:
                dic[word] += 1
        maxcount = 0
        most = ""
        for w in dic:
            if dic[w] > maxcount:
                maxcount = dic[w]
                most = w
        return most